### PR TITLE
README.md: Add npmjs badge

### PR DESCRIPTION
**Pull request by perlun**

### CODE REVIEW

**Comment by joliss** [comment]
I think this is too many badges. I'd prefer to just have the build badges!

**Comment by perlun** [comment]
> Hm, I think this is too many badges. I'd prefer to just have the build badges!

The point is not the badge, but the _link_ to the package on npmjs, i.e. https://www.npmjs.com/package/broccoli-merge-trees which the badge provides. Right now, you have to google this info yourself which I think is bad.

If you prefer a text link to the `npm` package, that's of course also fine. Feel free to add it yourself or let me know where you think it should go and I can amend the PR/open a new one.

**Comment by perlun** [comment]
Ping @joliss, did you see my followup message?

**Comment by joliss** [comment]
I'm not entirely convinced that the link provides much value, since the npm page still is mostly just another copy of the README.

**Comment by perlun** [comment]
@joliss 

> I'm not entirely convinced that the link provides much value, since the npm page still is mostly just another copy of the README.

You seem to miss my point.

People might find the repository here on GitHub, and see that you can `npm install` it - but don't know how to find the package page on npmjs.com. Or be too lazy too google it. I find it a reasonable courtesy to the users to provide this service.

Besides, the npmjs page shows a lot of other nice information than just the README:

- The most recent version of the package.
- The number of downloads (which is an important metric in singling out "serious" from "not-so-serious" packages)
- The dependencies and dependents - also a significant factor to look at when determining if a package is  "good" or "bad" dependency.

In the case of this specific package, the npmjs page gives a clear message - this is a good, serious package with a serious enough reputation for hundreds of other packages to make themselves dependent on it.

Simply put, I think it adds value. You are free to disagree, but it would be nice to hear some more counter-arguments in that case.
